extensions/axon/runtime/src/main/java/io/quarkus/axon/runtime/AxonRuntimeTemplate.java

```diff
@@ -0,0 +1,55 @@
+package io.quarkus.axon.runtime;
+
+import org.axonframework.config.Configuration;
+import org.jboss.logging.Logger;
+
+import io.quarkus.arc.runtime.BeanContainer;
+import io.quarkus.runtime.annotations.Template;
+
+@Template
+public class AxonRuntimeTemplate {
+
+    private static final Logger log = Logger.getLogger(AxonRuntimeTemplate.class);
+
+    public void setAxonBuildConfig(BeanContainer container, AxonRuntimeConfig axonRuntimeConfig) {
+        container.instance(AxonClientProducer.class).setAxonRuntimeConfig(axonRuntimeConfig);
+    }
+
+    public void registerSaga(BeanContainer container, Class<?> axonAnnotatedClass) {
```

Comment:
I think it would be preferable to have a single method that registers everything necessary with the `BeanContainer` much like is done in: https://github.com/quarkusio/quarkus/blob/7e9330df7f9b7f66887d94d15cd073379fff253a/extensions/vertx-web/runtime/src/main/java/io/quarkus/vertx/web/runtime/VertxWebRecorder.java#L46